index.js

```diff
@@ -460,7 +460,7 @@ function getEncodingHeader (encodings, request) {
       // consider the no-preference token as gzip for downstream compat
       // and x-gzip as an alias of gzip
       // ref.: [HTTP/1.1 RFC 7230 section 4.2.3](https://datatracker.ietf.org/doc/html/rfc7230#section-4.2.3)
-      .replace(/\*|x-gzip/g, 'gzip')
+      .replace(/\*|x-gzip/gu, 'gzip')
```

Comment:
Should we extract the regexp.in a separate variable to avoid initializing it evert time?

Reply:
Even though V8 may optimize objects inside functions (RegExp included), I always see an increase in performance after moving them out and using references

So I'd say yes, better to make sure it's not causing any degradations

Reply:
Updated in https://github.com/fastify/fastify-compress/pull/273/commits/73cfbad5e41eef17b9b0cb4649493503cc9a3d96